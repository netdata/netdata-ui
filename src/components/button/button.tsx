import React, { SyntheticEvent, FC } from "react"
import { StyledButton, StyledCircularProgress } from "./styled"
import { Icon } from "../icon"

export enum ButtonType {
  default = "default",
  noFill = "noFill",
}

export interface ButtonProps {
  id?: string
  ref?: React.MutableRefObject<HTMLInputElement>
  className?: string
  label?: string
  onClick?: (e: SyntheticEvent<HTMLButtonElement>) => void
  icon?: any
  isLoading?: boolean
  type?: ButtonType
  disabled?: boolean
}

export const Button: FC<ButtonProps> = ({
  label = "LABEL",
  onClick = () => {},
  icon = null,
  isLoading = false,
  type = ButtonType.default,
  disabled = false,
}: ButtonProps) => {
  return (
    <StyledButton
      label={label}
      onClick={onClick}
      // isLoading={isLoading}
      icon={isLoading ? <StyledCircularProgress /> : <Icon name={icon} />}
      type={type}
      disabled={disabled}
    />
  )
}
