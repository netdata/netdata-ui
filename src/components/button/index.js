export { Button } from "./button"
export { default as IconButton } from "./IconButton"
export { ButtonGroup } from "./buttonGroup"
