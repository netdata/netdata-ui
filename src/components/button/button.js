import React, { forwardRef } from "react"
import { StyledButton } from "./styled"
import { Icon } from "@/components/icon"
import { LoaderIcon } from "@/components/icon/components"
import Flex from "@/components/templates/flex"

export const Button = forwardRef(
  (
    {
      label,
      icon,
      flavour,
      isLoading,
      loadingLabel,
      onClick,
      textTransform = "firstLetter",
      iconColor,
      iconSize,
      iconWidth,
      iconHeight,
      children = label,
      ...rest
    },
    ref
  ) => (
    <StyledButton
      flavour={flavour}
      textTransform={textTransform}
      hasIcon={!!icon || isLoading}
      onClick={isLoading ? undefined : onClick}
      ref={ref}
      iconColor={iconColor}
      {...rest}
    >
      {isLoading && <LoaderIcon className="button-icon" />}
      {icon && !isLoading && (
        <Flex justifyContent="center" alignItems="center" width="auto" height="100%">
          <Icon
            size={iconSize}
            className={iconColor ? "button-icon__color" : "button-icon"}
            title={icon}
            name={icon}
            width={iconWidth}
            height={iconHeight}
          />
        </Flex>
      )}

      {!!children && <span>{(isLoading && loadingLabel) || children}</span>}
    </StyledButton>
  )
)

Button.defaultProps = {
  onClick: () => {},
  icon: null,
}
