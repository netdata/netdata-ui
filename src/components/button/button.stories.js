import React, { useState } from "react"
import { Button, IconButton, ButtonGroup } from "."
import { iconsList } from "@/components/icon"

const icons = Object.keys(iconsList)

export const Basic = args => <Button {...args} />

export const BaseIconButton = args => (
  <IconButton
    {...args}
    label=""
    cursor="pointer"
    icon="chevron_left"
    iconSize="small"
    tooltip="Previous"
  />
)

const radioButtonItems = [
  { label: "One", value: 1 },
  { label: "Two", value: 2 },
  { label: "Three", value: 3 },
]

export const RadioButtonGroup = args => {
  const [checked, setChecked] = useState(1)
  const onChange = value => setChecked(value)

  return (
    <ButtonGroup
      items={radioButtonItems.map(item => ({ ...args, ...item }))}
      checked={checked}
      onChange={onChange}
    />
  )
}

export default {
  component: Button,
  tags: ["autodocs"],
  args: {
    onClick: () => alert("clicked"),
    flavour: "default",
    id: "some-id",
    className: "some-classname",
    label: "My button",
    loadingLabel: "Loading button",
    icon: icons[0],
    isLoading: false,
    disabled: false,
    danger: false,
    warning: false,
    small: false,
    neutral: false,
  },
  argTypes: {
    flavour: {
      options: ["default", "hollow", "borderless"],
      control: { type: "radio" },
    },
    id: { control: "text" },
    className: { control: "text" },
    label: { control: "text" },
    loadingLabel: { control: "text" },
    icon: { options: icons, type: "select" },
    isLoading: { control: "boolean" },
    disabled: { control: "boolean" },
    danger: { control: "boolean" },
    warning: { control: "boolean" },
    small: { control: "boolean" },
    neutral: { control: "boolean" },
  },
}
