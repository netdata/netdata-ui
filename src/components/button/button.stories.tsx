import React from "react"
import { storiesOf } from "@storybook/react"
import { action } from "@storybook/addon-actions"
import { text, boolean, select } from "@storybook/addon-knobs"
import { readmeCleanup } from "../../../utils/readme"
import { Button } from "."

// @ts-ignore
import readme from "./README.md"
import { ButtonType } from "./button"
import { iconsList } from "../icon"

const buttonStory = storiesOf("COMPONENTS|Controls/Button", module)
const icons: string[] = Object.keys(iconsList)

const subData = {
  readme: {
    sidebar: readmeCleanup(readme),
  },
  jest: ["button.test.tsx"],
}

buttonStory.add(
  "Button Default",
  () => (
    <Button
      disabled={boolean("Disabled", false)}
      label={text("Label prop", "LABEL")}
      onClick={action("clicked")}
    />
  ),
  subData
)

buttonStory.add(
  "Button No Fill",
  () => (
    <Button
      disabled={boolean("Disabled", false)}
      type={ButtonType.noFill}
      onClick={action("clicked")}
      label={text("Label", "LABEL")}
    />
  ),
  subData
)

buttonStory.add(
  "Button Borderless",
  () => (
    <Button
      disabled={boolean("Disabled", false)}
      type={ButtonType.borderLess}
      onClick={action("clicked")}
      label={text("Label", "LABEL")}
    />
  ),
  subData
)

buttonStory.add(
  "Button Action",
  () => (
    <Button
      icon={select("name", icons, "plus")}
      disabled={boolean("Disabled", false)}
      onClick={action("clicked")}
    />
  ),
  subData
)

buttonStory.add(
  "Button Action No Fill",
  () => (
    <Button
      icon={select("name", icons, "plus")}
      disabled={boolean("Disabled", false)}
      onClick={action("clicked")}
      type={ButtonType.noFill}
    />
  ),
  subData
)
