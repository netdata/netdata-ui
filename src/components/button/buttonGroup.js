import React, { Children, isValidElement, cloneElement } from "react"
import Flex from "@/components/templates/flex"
import { Button } from "./button"

const getButtonGroupProps = (itemIndex, itemsLength) => {
  const groupFirst = itemIndex == 0
  const groupLast = itemIndex == itemsLength - 1

  return {
    groupFirst,
    groupLast,
    groupMiddle: !groupFirst && !groupLast,
  }
}

const Content = ({ children }) => {
  const renderedChildren = Children.toArray(children).length

  return (
    <>
      {Children.map(children, (child, index) => {
        if (isValidElement(child)) {
          const buttonGroupProps = getButtonGroupProps(index, renderedChildren)
          return cloneElement(child, buttonGroupProps)
        }

        return child
      })}
    </>
  )
}

const RadioButtons = ({ items, checked, buttonProps = {}, onChange }) => (
  <>
    {items.map(({ label, value }, index) => {
      const buttonGroupProps = getButtonGroupProps(index, items.length)
      return (
        <Button
          key={value}
          label={label}
          onClick={() => onChange(value)}
          {...(checked != value ? { flavour: "hollow" } : {})}
          {...buttonGroupProps}
          {...buttonProps}
        />
      )
    })}
  </>
)

export const ButtonGroup = ({ items, checked, onChange, children, buttonProps, ...props }) => (
  <Flex alignItems="center" {...props}>
    {items?.length ? (
      <RadioButtons items={items} checked={checked} onChange={onChange} buttonProps={buttonProps} />
    ) : (
      <Content>{children}</Content>
    )}
  </Flex>
)
