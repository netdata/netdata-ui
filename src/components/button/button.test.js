/**
 * @jest-environment jsdom
 */
import React from "react"
import { renderWithProviders, fireEvent } from "testUtilities"
import { Button } from "./button"

describe("Button states", () => {
  it("renders disabled", () => {
    const { container } = renderWithProviders(<Button label="Test prop text" disabled />)
    const button = container.firstChild
    expect(button).toHaveStyleRule("opacity", "0.4", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("pointer-events", "none", {
      modifier: "&&",
    })
  })

  it("renders a button with text being in default format", () => {
    const { container } = renderWithProviders(<Button label="Test prop text" />)
    const button = container.firstChild
    expect(button).toHaveStyleRule("text-transform", "lowercase", {
      modifier: "&& > span",
    })
    expect(button).toHaveStyleRule("text-transform", "uppercase", {
      modifier: "&& > span::first-letter",
    })
  })

  it("renders a button with text that has no text-transform", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" textTransform="none" />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("text-transform", "none", {
      modifier: "&& > span",
    })
  })

  it("renders a button with text being capitalized", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" textTransform="capitalize" />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("text-transform", "capitalize", {
      modifier: "&& > span",
    })
  })

  it("renders a button with text being in uppercase", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" textTransform="uppercase" />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("text-transform", "uppercase", {
      modifier: "&& > span",
    })
  })

  it("renders a button with text being in lowercase", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" textTransform="lowercase" />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("text-transform", "lowercase", {
      modifier: "&& > span",
    })
  })

  it("renders a button with text being in full-width", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" textTransform="fullWidth" />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("text-transform", "full-width", {
      modifier: "&& > span",
    })
  })

  it("renders a text with only one capital letter, the first one", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" textTransform="firstLetter" />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("text-transform", "lowercase", {
      modifier: "&& > span",
    })
    expect(button).toHaveStyleRule("text-transform", "uppercase", {
      modifier: "&& > span::first-letter",
    })
  })

  it("renders loading text", () => {
    const { getByText } = renderWithProviders(
      <Button label="Test prop text" loadingLabel="loading..." isLoading />
    )
    expect(getByText(/loading/)).toBeInTheDocument()
  })

  it("renders with icon", () => {
    const { container, getByText, getByTitle } = renderWithProviders(
      <Button label="Test prop text" icon="plus" />
    )
    const button = container.firstChild

    expect(getByText(/prop text/)).toBeInTheDocument()
    expect(button).toHaveStyleRule("margin-left", "2px", {
      modifier: "&& > span",
    })
    expect(getByTitle("plus")).toBeInTheDocument()
  })

  it("renders only icon", () => {
    const { container } = renderWithProviders(<Button flavour="hollow" icon="plus" />)
    const button = container.firstChild
    expect(button).toMatchSnapshot()
  })

  it("renders smaller only icon", () => {
    const { container } = renderWithProviders(<Button flavour="hollow" icon="plus" small />)
    const button = container.firstChild
    expect(button).toMatchSnapshot()
  })

  it("renders loading icon", () => {
    const { container, getByText, queryByTitle } = renderWithProviders(
      <Button label="Test prop text" icon="plus" isLoading />
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(/prop text/)).toBeInTheDocument()
    expect(queryByTitle("plus")).not.toBeInTheDocument()
  })

  it("is clickable", () => {
    const onClickFn = jest.fn()
    const { getByText } = renderWithProviders(<Button label="Test prop text" onClick={onClickFn} />)

    fireEvent.click(getByText(/prop text/))
    expect(onClickFn).toBeCalled()
  })
})

describe("Default Button", () => {
  it("renders", () => {
    const { container, getByText } = renderWithProviders(
      <Button label="Test prop text" textTransform="fullWidth" />
    )

    expect(getByText(/prop text/)).toBeInTheDocument()

    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("background-color", "#00AB44", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#00AB44", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#FFFFFF", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("opacity", "1", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("pointer-events", "auto", {
      modifier: "&&",
    })
  })

  it("renders for danger", () => {
    const { container } = renderWithProviders(<Button label="Test prop text" danger />)
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#FF4136", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#FF4136", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#FFFFFF", {
      modifier: "&&",
    })
  })

  it("renders for warning", () => {
    const { container } = renderWithProviders(<Button label="Test prop text" warning />)
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#F9A825", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#F9A825", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#FFFFFF", {
      modifier: "&&",
    })
  })

  it("renders neutral", () => {
    const { container } = renderWithProviders(<Button label="Test prop text" neutral />)
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#536775", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#536775", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#FFFFFF", {
      modifier: "&&",
    })
  })
})

describe("Hollow Button", () => {
  it("renders", () => {
    const { container, getByText } = renderWithProviders(
      <Button label="Test prop text" flavour="hollow" />
    )

    expect(getByText(/prop text/)).toBeInTheDocument()

    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#00AB44", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#00AB44", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("opacity", "1", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("pointer-events", "auto", {
      modifier: "&&",
    })
  })

  it("renders for danger", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" flavour="hollow" danger />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#FF4136", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#FF4136", {
      modifier: "&&",
    })
  })

  it("renders for warning", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" flavour="hollow" warning />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#F9A825", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#F9A825", {
      modifier: "&&",
    })
  })

  it("renders for neutral", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" flavour="hollow" neutral />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "#8F9EAA", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#8F9EAA", {
      modifier: "&&",
    })
  })
})

describe("Borderless Button", () => {
  it("renders", () => {
    const { container, getByText } = renderWithProviders(
      <Button label="Test prop text" flavour="borderless" />
    )

    expect(getByText(/prop text/)).toBeInTheDocument()

    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#00AB44", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("opacity", "1", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("pointer-events", "auto", {
      modifier: "&&",
    })
  })

  it("renders for danger", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" flavour="borderless" danger />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#FF4136", {
      modifier: "&&",
    })
  })

  it("renders for warning", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" flavour="borderless" warning />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#F9A825", {
      modifier: "&&",
    })
  })

  it("renders for neutral", () => {
    const { container } = renderWithProviders(
      <Button label="Test prop text" flavour="borderless" neutral />
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("border-color", "rgba(255,255,255,0.0)", {
      modifier: "&&",
    })
    expect(button).toHaveStyleRule("color", "#536775", {
      modifier: "&&",
    })
  })
})
