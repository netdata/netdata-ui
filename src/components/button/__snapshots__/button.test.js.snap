// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Borderless Button renders 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 500;
  font-size: 14px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: rgba(255,255,255,0.0);
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(255,255,255,0.0);
  border-radius: 4px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0 > span {
  text-transform: lowercase;
  margin-left: 0px;
}

.c0.c0 > span::first-letter {
  text-transform: uppercase;
}

.c0.c0:hover {
  border-color: rgba(255,255,255,0.0);
  background-color: rgba(255,255,255,0.0);
  color: #00CD51;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0:hover .button-icon {
  fill: #00CD51;
}

.c0.c0:active {
  border-color: rgba(255,255,255,0.0);
  background-color: rgba(255,255,255,0.0);
  color: #00CD51;
  box-shadow: inset 0 4px 4px rgba(0,0,0,0.25);
}

.c0.c0:focus {
  outline: none;
}

.c0.c0 .button-icon {
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #00AB44;
  width: 24px;
}

.c0.c0 .path {
  stroke: #00AB44;
}

<button
  class="c0"
>
  <span>
    Test prop text
  </span>
</button>
`;

exports[`Button states renders loading icon 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 500;
  font-size: 14px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #00AB44;
  color: #F6F7F7;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 4px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0 > span {
  text-transform: lowercase;
  margin-left: 4px;
}

.c0.c0 > span::first-letter {
  text-transform: uppercase;
}

.c0.c0:hover {
  border-color: #42B861;
  background-color: #42B861;
  color: #F6F7F7;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0:hover .button-icon {
  fill: #F6F7F7;
}

.c0.c0:active {
  border-color: #00CD51;
  background-color: #00CD51;
  color: #F6F7F7;
  box-shadow: inset 0 4px 4px rgba(0,0,0,0.25);
}

.c0.c0:focus {
  outline: none;
}

.c0.c0 .button-icon {
  height: 16px;
  width: 16px;
  fill: #F6F7F7;
}

.c0.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #F6F7F7;
  width: 24px;
}

.c0.c0 .path {
  stroke: #F6F7F7;
}

.c1 {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #FFFFFF;
  width: 24px;
}

.c1 .path {
  stroke: #FFFFFF;
}

<button
  class="c0"
>
  <svg
    class="c1 button-icon"
    version="1.1"
    viewBox="0 0 21 17"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      class="path"
      fill="none"
      fill-rule="evenodd"
      stroke="none"
      stroke-width="1"
    >
      <path
        d="M2,1 C8.25086152,1 11.9367136,1 13.0575562,1 C14.73882,1 19.6834591,2 19.9614325,7.72050108 C20.239406,13.4410022 15.7459591,15.1224845 13.6463763,15.1224845 C12.2466545,15.1224845 10.0279195,15.1224845 6.9901715,15.1224845 L2,1 Z"
        id="Path-2"
        stroke-width="2"
      />
    </g>
  </svg>
  <span>
    Test prop text
  </span>
</button>
`;

exports[`Button states renders only icon 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 500;
  font-size: 14px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: rgba(255,255,255,0.0);
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 4px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0 > span {
  text-transform: lowercase;
  margin-left: 4px;
}

.c0.c0 > span::first-letter {
  text-transform: uppercase;
}

.c0.c0:hover {
  border-color: #00AB44;
  background-color: #DDFFEB;
  color: #00AB44;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0:hover .button-icon {
  fill: #00AB44;
}

.c0.c0:active {
  border-color: #00AB44;
  background-color: #DDFFEB;
  color: #00AB44;
  box-shadow: inset 0 4px 4px rgba(0,0,0,0.25);
}

.c0.c0:focus {
  outline: none;
}

.c0.c0 .button-icon {
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #00AB44;
  width: 24px;
}

.c0.c0 .path {
  stroke: #00AB44;
}

.c2 {
  height: 16px;
  width: 16px;
  opacity: 1;
  pointer-events: unset;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 100%;
  color: #526161;
}

<button
  class="c0"
>
  <div
    class="c1"
    height="100%"
    width="auto"
  >
    <svg
      class="c2 button-icon"
      title="plus"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
</button>
`;

exports[`Button states renders smaller only icon 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 500;
  font-size: 12px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 4px 12px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: rgba(255,255,255,0.0);
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 4px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0 > span {
  text-transform: lowercase;
  margin-left: 4px;
}

.c0.c0 > span::first-letter {
  text-transform: uppercase;
}

.c0.c0:hover {
  border-color: #00AB44;
  background-color: #DDFFEB;
  color: #00AB44;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0:hover .button-icon {
  fill: #00AB44;
}

.c0.c0:active {
  border-color: #00AB44;
  background-color: #DDFFEB;
  color: #00AB44;
  box-shadow: inset 0 4px 4px rgba(0,0,0,0.25);
}

.c0.c0:focus {
  outline: none;
}

.c0.c0 .button-icon {
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #00AB44;
  width: 24px;
}

.c0.c0 .path {
  stroke: #00AB44;
}

.c2 {
  height: 16px;
  width: 16px;
  opacity: 1;
  pointer-events: unset;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 100%;
  color: #526161;
}

<button
  class="c0"
>
  <div
    class="c1"
    height="100%"
    width="auto"
  >
    <svg
      class="c2 button-icon"
      title="plus"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
</button>
`;

exports[`Default Button renders 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 500;
  font-size: 14px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #00AB44;
  color: #F6F7F7;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 4px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0 > span {
  text-transform: full-width;
  margin-left: 0px;
}

.c0.c0:hover {
  border-color: #42B861;
  background-color: #42B861;
  color: #F6F7F7;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0:hover .button-icon {
  fill: #F6F7F7;
}

.c0.c0:active {
  border-color: #00CD51;
  background-color: #00CD51;
  color: #F6F7F7;
  box-shadow: inset 0 4px 4px rgba(0,0,0,0.25);
}

.c0.c0:focus {
  outline: none;
}

.c0.c0 .button-icon {
  height: 16px;
  width: 16px;
  fill: #F6F7F7;
}

.c0.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #F6F7F7;
  width: 24px;
}

.c0.c0 .path {
  stroke: #F6F7F7;
}

<button
  class="c0"
>
  <span>
    Test prop text
  </span>
</button>
`;

exports[`Hollow Button renders 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 500;
  font-size: 14px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: rgba(255,255,255,0.0);
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 4px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0 > span {
  text-transform: lowercase;
  margin-left: 0px;
}

.c0.c0 > span::first-letter {
  text-transform: uppercase;
}

.c0.c0:hover {
  border-color: #00AB44;
  background-color: #DDFFEB;
  color: #00AB44;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.c0:hover .button-icon {
  fill: #00AB44;
}

.c0.c0:active {
  border-color: #00AB44;
  background-color: #DDFFEB;
  color: #00AB44;
  box-shadow: inset 0 4px 4px rgba(0,0,0,0.25);
}

.c0.c0:focus {
  outline: none;
}

.c0.c0 .button-icon {
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #00AB44;
  width: 24px;
}

.c0.c0 .path {
  stroke: #00AB44;
}

<button
  class="c0"
>
  <span>
    Test prop text
  </span>
</button>
`;
